### ISSUE TITLE
Remove lkm

### ISSUE THREAD

**Issue by rennergade** [Lind-Project/lind_project#316]
A while back we added a loadable kernel module (LKM) to attempt to speed up fork(). We eventually decided that dealing with the potential bugs involved as well as loading the module wasn't really worth the small overall performance gains (due to fork() being a small % of our total application time).

The code to utilize the lkm is still in native_client. Let's remove that to avoid any confusion later on.

@yizhuoliang can you remove the lkm code from native_client?

Basically you just need to search for `void CheckForLkm(void)` and the bool `use_lkm` and remove the paths in any if statements associated with them.

**Comment by JustinCappos** [comment]
When we start mapping memory between cages, might we need a similar mechanism?

**Comment by JustinCappos** [comment]
Just to clarify my prior comment, I mean mapping memory between cages as a way to pass parameters, objects which are passed by reference, etc. for API calls.

**Comment by rennergade** [comment]
@JustinCappos I somehow missed these comments. Some thoughts

1. We're not going to delete the actual LKM code, just want to remove the pieces that meshed it into NaCl for fork() if we need it later for another purpose it won't be difficult to add back in.
2. What will be difficult is getting everything to work correctly, as of now there seems to still be several bugs and its very complicated, but if we decide its worth it we can address that
3. We made the LKM to do CoW for fork in a way that would mirrorPOSIX fork() semantics. I'm not sure this would actually give us an advantage over mmap for what you're referring to.

**Comment by JustinCappos** [comment]
Sounds good.  Feel free to remove it.  

Just keep in mind we may need some customize shared memory semantics in the future, even if only for microbenchmarking / testing...

**Comment by rennergade** [comment]
OK @yizhuoliang  this is good to start on.

**Comment by rennergade** [comment]
Fix by @yizhuoliang
